DFC.ServiceTaxonomy.ApiFunction/Function/Execute.cs

```diff
@@ -52,6 +52,13 @@ public Execute(IOptionsMonitor<ServiceTaxonomyApiSettings> serviceTaxonomyApiSet
                 var environment = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_ENVIRONMENT");
                 log.LogInformation($"Function has been triggered in {environment} environment.");
 
+                var host = req.Headers["X-Forwarded-Host"].ToString();
+
+                if (string.IsNullOrWhiteSpace(host))
+                    throw new ArgumentNullException("X-Forwarded-Host not present.");
```

Comment:
Can we throw a more appropriate exception?
It's not really an argument and it also gets thrown if it's whitespace.

Reply:
Could go for ApiFunction.InternalServerError? We wouldn't want to serve content without the host being present would we?